flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/plan/nodes/physical/common/CommonPhysicalJoin.scala

```diff
@@ -80,4 +81,14 @@ abstract class CommonPhysicalJoin(
       .item("select", getRowType.getFieldNames.mkString(", "))
   }
 
+  def getUniqueKeys(input: RelNode, keys: Array[Int]): List[Array[Int]] = {
```

Comment:
getUpsertKeys ?

Reply:
this method was moved from `StreamPhysicalJoin`, so I didn't change the original name (there're several related methods named `xxUniqueKeys`, maybe better refact this in another pr)